src/deploy/functions/runtimes/node/triggerParser.js

```diff
@@ -21,7 +21,7 @@ async function loadModule(packageDir) {
   try {
     return require(packageDir);
   } catch (e) {
-    if (e.code === "ERR_REQUIRE_ESM") {
+    if (e.code === "ERR_REQUIRE_ESM" && e.code !== "ERR_REQUIRE_ASYNC_MODULE") {
```

Comment:
This condition seems wrong. `e.code !== "ERR_REQUIRE_ASYNC_MODULE"` is always true if `e.code === "ERR_REQUIRE_ESM"`.